internal/compiler/object_tree.rs

```diff
@@ -769,7 +776,32 @@ pub struct ListViewInfo {
 
 #[derive(Debug, Clone)]
 /// If the parent element is a repeated element, this has information about the models
-pub struct RepeatedElementInfo {
+pub enum RepeatedElementInfo {
```

Comment:
Instead of storing an Option<RepeatedElementInfo> I believe the code will become even simpler if you add a None variant to the enum. In this case the enum is always used in the context of the Option.